lib/resty/redis.lua

```diff
@@ -76,6 +76,17 @@ function _M.set_timeout(self, timeout)
 end
 
 
+function _M.set_timeouts(self, connect_timeout, send_timeout, read_timeout)
+    local sock = rawget(self, "_sock")
+    if not sock then
+        error("not initialized", 2)
+        return
+    end
+
+    return sock:settimeouts(connect_timeout, send_timeout, read_timeout)
```

Comment:
I don't think the `settimeouts` method returns anything, better avoid returning here. Also this is a tail-call, which is bad for debugging.

Reply:
it's just a compatible with `function _M.set_timeout`
according to [src](https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_socket_tcp.c#L3232),I think 
settimeouts returns 0 when success ,yet otherwise returns luaL_error code when timeout number out-of-range of 32bit int

Reply:
@php-cpm Then that function needs updating too (though for a separate issue and PR). These settimeout and settimouts functions are not meant to have any meaningful return values at all. See

https://github.com/openresty/lua-nginx-module#tcpsocksettimeouts

and

https://github.com/openresty/lua-nginx-module#tcpsocksettimeout

Reply:
fine, how do they look like now?